frontend/packages/console-app/src/redux/actions/cloud-shell-actions.ts

```diff
@@ -1,13 +1,31 @@
+import { useCallback } from 'react';
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore: FIXME missing exports due to out-of-sync @types/react-redux version
+import { useDispatch, Dispatch } from 'react-redux';
 import { RootState } from '@console/internal/redux';
-import { Dispatch } from 'react-redux';
 import { action, ActionType } from 'typesafe-actions';
 import { isCloudShellExpanded, isCloudShellActive } from '../reducers/cloud-shell-selectors';
 
 export enum Actions {
   SetCloudShellExpanded = 'setCloudShellExpanded',
   SetCloudShellActive = 'setCloudShellActive',
+  SetCloudShellCommand = 'SetCloudShellCommand',
 }
 
+const cloudShellCommand = (command: string | null) =>
+  action(Actions.SetCloudShellCommand, { command });
+
+export const useCloudShellCommandDispatch = (): ((command: string | null) => void) => {
```

Comment:
Not sure if this should be part of this file. Its actually an action dispatcher and not an action creator.

Reply:
we can create a new file `cloud-shell-action-dispatcher` under the actions directory. WDYT?

Reply:
I think if are keeping the separate component to handle execution of the command then this can be part of this component itself. I guess my question is why have a separate dispatcher utility when it is just dispatching an action. The components can handle this code directly.

Reply:
So it's easy to use/import the utility since we don't have redux types up to date.

Reply:
First off, we should definitely upgrade the redux types. Its been like this for so long now. And secondly my concern with keeping separation of concerns because dispatching an action is usually done from a component and should be part of component logic. If not then we can have a utility hook in utils folder related to `cloud-shell` that any component can use. Keeping this here along with actions and reducers feels a bit odd.

Reply:
On a second thought, I guess it doesn't really matters. You can keep this in a separate file named `cloud-shell-dispatchers.ts`. We can maybe start to use this pattern whenever we want to reuse a complex dispatch logic anywhere.